modules/infoserv/main.c

```diff
@@ -570,7 +667,35 @@ mod_init(struct module *const restrict m)
 static void
 mod_deinit(const enum module_unload_intent ATHEME_VATTR_UNUSED intent)
 {
-
+	struct infoserv_persist_record *rec = smalloc(sizeof *rec);
+	rec->version        = INFOSERV_PERSIST_VERSION;
+	rec->logon_info     = logon_info;
+	rec->operlogon_info = operlogon_info;
+
+	mowgli_global_storage_put(INFOSERV_PERSIST_STORAGE_NAME, rec);
+
+	del_conf_item("LOGONINFO_COUNT", &infoserv->conf_table);
+	del_conf_item("LOGONINFO_REVERSE", &infoserv->conf_table);
+	del_conf_item("LOGONINFO_SHOW_METADATA", &infoserv->conf_table);
+
+	hook_del_user_add(hook_user_add);
+	hook_del_user_oper(hook_user_oper);
+	hook_del_operserv_info(osinfo_hook);
+	hook_del_db_write(write_infodb);
+
+	db_unregister_type_handler(DB_TYPE_LOGONINFO);
+	db_unregister_type_handler(DB_TYPE_LOGONINFO_OPER);
+
+	service_unbind_command(infoserv, &is_help);
+	service_unbind_command(infoserv, &is_post);
+	service_unbind_command(infoserv, &is_del);
+	service_unbind_command(infoserv, &is_odel);
+	service_unbind_command(infoserv, &is_move);
+	service_unbind_command(infoserv, &is_omove);
+	service_unbind_command(infoserv, &is_list);
+	service_unbind_command(infoserv, &is_olist);
```

Comment:
These are unnecessary because `service_delete()` now [clears out the command trie](https://github.com/atheme/atheme/blob/master/libathemecore/servtree.c#L382-L383) itself (commit 2a9d68b63285ce77d157).